prboom2/src/dsda/console.c

```diff
@@ -1654,7 +1654,7 @@ static dboolean console_StateSetMisc2(const char* command, const char* args) {
 
 static dboolean console_StateSetArgs1(const char* command, const char* args) {
   int id;
-  statearg_t value;
```

Comment:
The whole point of having a `statearg_t` is to use it when we're dealing with state arguments, for consistency and clarity. Let's rather change the typedef for `statearg_t` to be a `long long`.

Reply:
`statearg_t` is  `int64_t` because it's read from a file, right?  `long long` is 64 bits on most platforms, but it's not guaranteed, which is why this generates a warning in the first place.  If that's still OK, I can go ahead and change it.

Reply:
`statearg_t` are dehacked parameters, which are parsed from plain text and set with lines like this: `deh_state.state->args[0] = (statearg_t)value;`. Incidentally dehacked parameters are all 32 bit integers. The reason this is 64 bits is because some arguments are translated from dehacked values into port-specific flags, which can use higher bits. E.g., `states[i].args[1] = deh_translate_bits(states[i].args[1], deh_mobjflags_mbf21);`. So as long as this field has at least 64 bits it should all be fine.

That's assuming it won't introduce extra warnings when converting types here:
```c
void A_AddFlags(mobj_t* actor)
{
  uint64_t flags, flags2;

  if (!mbf21 || !actor)
    return;

  flags  = actor->state->args[0];
  flags2 = actor->state->args[1];

  actor->flags  |= flags;
  actor->flags2 |= flags2;
}
```

Reply:
It should be fine, I don't think anyone turns on warnings about implicit integer conversions in C because it's so prolific in practice.